types/field_type.go

```diff
@@ -40,6 +40,9 @@ type FieldType struct {
 	Collate string
 	// Elems is the element list for enum and set type.
 	Elems []string
+	// NegativeUnsignedBeZero indicates that unsigned type with negative number will be cast 0 instead of Big number.
+	// it's only used in union implicitly cast.
+	NegativeUnsignedBeZero bool
```

Comment:
Please take more time to find a way to avoid adding this field.

Reply:
I think we should move this field to the cast function.